supabase/migrations/20251209150000_spark_send_quotes.sql

```diff
@@ -0,0 +1,348 @@
+        message = format('Failed to complete quote with id %s.', v_quote.id),
+        detail = format('Quote is in state %s but must be in PENDING state.', v_quote.state);
+  end if;
+
+  -- Update quote to completed state
+  update wallet.spark_send_quotes
+  set
+    state = 'COMPLETED',
+    payment_preimage = p_payment_preimage,
+    spark_transfer_id = p_spark_transfer_id,
+    fee = p_fee,
+    version = version + 1
+  where id = p_quote_id
+  returning * into v_quote;
+
+  -- Update transaction to completed state
+  update wallet.transactions
+  set
+    state = 'COMPLETED',
+    acknowledgment_status = 'pending',
+    completed_at = now(),
+    encrypted_transaction_details = p_encrypted_transaction_details,
+    transaction_details = jsonb_build_object('sparkTransferId', p_spark_transfer_id)
+  where id = v_quote.transaction_id;
+
+  return v_quote;
+end;
+$function$;
+
+-- =============================================================================
+-- Function: fail_spark_send_quote
+-- Marks a spark send quote as failed and updates the transaction
+-- =============================================================================
+
+create or replace function wallet.fail_spark_send_quote(
+  p_quote_id uuid,
+  p_failure_reason text
+)
+returns wallet.spark_send_quotes
+language plpgsql
+security invoker
+set search_path = ''
+as $function$
+declare
+  v_quote wallet.spark_send_quotes;
+begin
+  -- Get the quote with a lock to prevent race conditions
+  select * into v_quote
+  from wallet.spark_send_quotes
+  where id = p_quote_id
+  for update;
+
+  if v_quote is null then
+    raise exception
+      using
+        hint = 'NOT_FOUND',
+        message = format('Quote with id %s not found.', p_quote_id);
+  end if;
+
+  -- If already failed, return the quote (idempotent)
+  if v_quote.state = 'FAILED' then
+    return v_quote;
+  end if;
+
+  -- Only PENDING quotes can be failed
+  if v_quote.state != 'PENDING' then
+    raise exception
+      using
+        hint = 'INVALID_STATE',
+        message = format('Failed to fail quote with id %s.', v_quote.id),
+        detail = format('Quote is in state %s but must be in PENDING state.', v_quote.state);
+  end if;
+
+  -- Update quote to failed state
+  update wallet.spark_send_quotes
+  set
+    state = 'FAILED',
+    failure_reason = p_failure_reason,
+    version = version + 1
+  where id = p_quote_id
+  returning * into v_quote;
+
+  -- Update transaction to failed state
+  update wallet.transactions
+  set
+    state = 'FAILED',
+    failed_at = now()
+  where id = v_quote.transaction_id;
+
+  return v_quote;
+end;
+$function$;
+
+-- =============================================================================
+-- Cron Jobs: Cleanup spark quotes
+-- =============================================================================
+
+-- Cleanup expired and paid spark receive quotes every day at midnight
+select cron.schedule('cleanup-spark-receive-quotes', '0 0 * * *', $$
```

Comment:
we should create indexes for (state, created_at) for both of these tables to make these delete queries fast